airbyte-webapp/src/components/ui/NextTable/NextTable.tsx

```diff
@@ -1,43 +1,72 @@
-import { ColumnDef, flexRender, useReactTable, getCoreRowModel } from "@tanstack/react-table";
+import { ColumnDef, flexRender, useReactTable, getCoreRowModel, ColumnSort } from "@tanstack/react-table";
 import classNames from "classnames";
 import { PropsWithChildren } from "react";
 
 import styles from "./NextTable.module.scss";
 
-export interface TableProps<TData> {
-  data: TData[];
-  columns: Array<ColumnDef<TData>>;
-  onClickRow?: (data: unknown) => void;
+type TData<T> = T & {
+  error?: unknown;
+};
+
+export interface TableProps<T> {
   className?: string;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  columns: Array<ColumnDef<TData<T>, any>>;
```

Comment:
Don't love this any but unsure if we can do anything about it. It's fine, just would like it if it was unnecessary.

Reply:
I left a comment in the `NextTable` component. We tried yesterday with @dizel852 to get rid of `any` here, but we didn't find a way. It also seems to be type safely, since the `useReactTable` `options.columns` itself expects `Array<ColumnDef<T, any>>`

Reply:
Confirm, double checked that